src/lib/UserConfig.ts

```diff
@@ -0,0 +1,16 @@
+import fs = require("fs");
+import os = require("os");
+
+const homedir = os.homedir();
+const configModule = `${homedir}/.newsletter-cli/config`;
+
+export class UserConfig {
+  public load() {
```

Comment:
please define the expected type of the config data. Something like:

```
export class UserConfig {
  private _config: unknown;
  constructor(path: string) {
     this._config = require(string);
  }
  
  getTemplateSubject(): string {
   const templateSubject = _config.template_subject;
    if (typeof templateSubject !== 'string') {
       throw new Error(...);
    }
    return templateSubject;
  }
};
```

Then the path can be resolved in index.ts and provided down to the classes that need the config.

Reply:
OK, that adjustment is pretty straightforward. Another question: is it fine to pass the entire instantiated `UserConfig` object down through the various constructors, and let the destination class call the methods on that object? E.g., the `Newsletter` class would call `getTemplateSubject()`, etc? I think that will be cleaner, and still provide the type checking you want for the config data.

Reply:
yes, that is what I meant

Reply:
OK, I think this is complete per your request.